packages/@aws-cdk/aws-codepipeline-actions/test/codestar-connections/codestar-connections-source-action.test.ts

```diff
@@ -148,27 +148,155 @@ describe('CodeStar Connections source Action', () => {
       ],
     });
 
+  });
+
+  test('exposes variables', () => {
+    const stack = new Stack();
+
+    createBitBucketAndCodeBuildPipeline(stack, {
+      triggerOnPush: false,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodePipeline::Pipeline', {
+      'Stages': [
+        {
+          'Name': 'Source',
+        },
+        {
+          'Name': 'Build',
+          'Actions': [
+            {
+              'Name': 'CodeBuild',
+              'Configuration': {
+                'EnvironmentVariables': '[{"name":"CommitId","type":"PLAINTEXT","value":"#{Source_BitBucket_NS.CommitId}"}]',
+              },
+            },
+          ],
+        },
+      ],
+    });
 
   });
+
+  test('exposes variables with custom namespace', () => {
+    const stack = new Stack();
+
+    createBitBucketAndCodeBuildPipeline(stack, {
+      triggerOnPush: false,
+      variablesNamespace: 'kornicameister',
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodePipeline::Pipeline', {
+      'Stages': [
+        {
+          'Name': 'Source',
+          'Actions': [
+            {
+              'Name': 'BitBucket',
+              'Namespace': 'kornicameister',
+            },
+          ],
+        },
+        {
+          'Name': 'Build',
+          'Actions': [
+            {
+              'Name': 'CodeBuild',
+              'Configuration': {
+                'EnvironmentVariables': '[{"name":"CommitId","type":"PLAINTEXT","value":"#{kornicameister.CommitId}"}]',
+              },
+            },
+          ],
+        },
+      ],
+    });
+
+
+  });
+
+  test('fail if variable from unused action is referenced', () => {
+    const stack = new Stack();
+
+    createUnreferencedVariablePipeline(stack);
+
+    expect(() => {
+      SynthUtils.synthesize(stack);
+    }).toThrow(/Cannot reference variables of action 'BitBucketUnused', as that action was never added to a pipeline/);
+
+  });
+
+  test('fail if variable from unused action with custom namespace is referenced', () => {
+    const stack = new Stack();
+
+    createUnreferencedVariablePipeline(stack, 'kornicameister');
+
+    expect(() => {
+      SynthUtils.synthesize(stack);
+    }).toThrow(/Cannot reference variables of action 'BitBucketUnused', as that action was never added to a pipeline/);
+
+  });
+
 });
 
-function createBitBucketAndCodeBuildPipeline(stack: Stack, props: Partial<cpactions.BitBucketSourceActionProps>): void {
+function createUnreferencedVariablePipeline(stack: Stack, variablesNamespace?: string) {
```

Comment:
Can you put this new function below the existing `createBitBucketAndCodeBuildPipeline()`?